nixos/modules/services/web-apps/lasuite-docs.nix

```diff
@@ -0,0 +1,576 @@
+          };
+
+          DB_NAME = mkOption {
+            type = types.str;
+            default = "lasuite-docs";
+            description = "Name of the database";
+          };
+
+          DB_USER = mkOption {
+            type = types.str;
+            default = "lasuite-docs";
+            description = "User of the database";
+          };
+
+          DB_HOST = mkOption {
+            type = types.nullOr types.str;
+            default = if cfg.database.createLocally then "/run/postgresql" else null;
+            description = "Host of the database";
+          };
+
+          REDIS_URL = mkOption {
+            type = types.nullOr types.str;
+            default =
+              if cfg.redis.createLocally then
+                "unix://${config.services.redis.servers.lasuite-docs.unixSocket}?db=1"
+              else
+                null;
+            description = "URL of the redis backend";
+          };
+
+          CELERY_BROKER_URL = mkOption {
+            type = types.nullOr types.str;
+            default =
+              if cfg.redis.createLocally then
+                "unix://${config.services.redis.servers.lasuite-docs.unixSocket}?db=2"
+              else
+                null;
+            description = "URL of the redis backend for celery";
+          };
+        };
+      };
+      default = { };
+      example = ''
+        {
+          DJANGO_ALLOWED_HOSTS = "*";
+        }
+      '';
+      description = ''
+        Configuration options of docs.
+
+        See https://github.com/suitenumerique/docs/blob/v${cfg.backendPackage.version}/docs/env.md
+
+        `REDIS_URL` and `CELERY_BROKER_URL` are set if `services.lasuite-docs.redis.createLocally` is true.
+        `DB_NAME`, `DB_USER` and `DB_HOST` are set if `services.lasuite-docs.database.createLocally` is true.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path to environment file.
+
+        This can be useful to pass secrets to docs via tools like `agenix` or `sops`.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.lasuite-docs = {
+      description = "Docs from SuiteNumérique";
+      after =
+        [ "network.target" ]
+        ++ (optional cfg.database.createLocally "postgresql.service")
+        ++ (optional cfg.redis.createLocally "redis-lasuite-docs.service");
+      wants =
+        (optional cfg.database.createLocally "postgresql.service")
+        ++ (optional cfg.redis.createLocally "redis-lasuite-docs.service");
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        if [ ! -f .version ]; then
+          touch .version
+        fi
+
+        if [ "${cfg.backendPackage.version}" != "$(cat .version)" ]; then
+          ${getExe cfg.backendPackage} migrate
+          ${getExe cfg.backendPackage} collectstatic --no-input --clear
```

Comment:
Why should we run collectstatic at runtime (in the module) and not at buildtime (in the package)?

Reply:
There will be no relevant changes between build and runtime, and we can reuse build time results for most/all consumers, which we can't do for runtime.

Whether you push the results into a dedicated `static` output depends on whether that simplifies the resulting configuration.

I would therefore default to your first proposal.